python/ray/rllib/utils/filter.py

```diff
@@ -5,19 +5,39 @@
 import numpy as np
 
 
-class NoFilter(object):
-    def __init__(self):
+class BaseFilter(object):
+
+    def update(self, other, *args, **kwargs):
+        """Given other filter, update self with "new state" from
+        other filter. This may be in form of a buffer.
+        """
+        raise NotImplementedError
+
+    def copy(self):
+        """Creates a new object with same state as self. Returns copy."""
+        raise NotImplementedError
+
+    def sync(self, other):
```

Comment:
why not set `filter = other.copy()`?

Reply:
For async reasons... `lock_wrap` takes the `MeanStdFilter` and passes it to the other thread. Changing the reference won't update the object in `lock_wrap`. Suggestions welcome here though.